File: KITE-Mediasoup-Test/src/main/java/io/cosmosoftware/kite/mediasoup/steps/JoinVideoCallStep.java
package io.cosmosoftware.kite.mediasoup.steps;

import io.cosmosoftware.kite.exception.KiteTestException;
import io.cosmosoftware.kite.mediasoup.pages.MediasoupPage;
import io.cosmosoftware.kite.steps.TestStep;
import org.openqa.selenium.WebDriver;

public class JoinVideoCallStep extends TestStep {

   
  private final String url;

  
  public JoinVideoCallStep(WebDriver webDriver, String url) {
    super(webDriver);
    this.url = url;
  }
  
  @Override
  public String stepDescription() {
    return "Open " + url;
  }
  
  @Override
  protected void step() throws KiteTestException {
    final MediasoupPage mediasoupPage = new MediasoupPage(this.webDriver, logger);
    mediasoupPage.load(url);
  }
}
  

File: KITE-Mediasoup-Test/js/steps/JoinVideoCallStep.js
const {TestStep} = require('kite-common');
const {mediasoupPage}  = require('../pages');

/**
 * Class: JoinVideoCallStep
 * Extends: TestStep
 * Description:
 */
class JoinVideoCallStep extends TestStep {
  constructor(kiteBaseTest, url) {
    super();
    this.driver = kiteBaseTest.driver;
    this.timeout = kiteBaseTest.timeout;
    this.url = url;
    this.uuid = kiteBaseTest.uuid;
    this.page = kiteBaseTest.page;
  }

  stepDescription() {
    return 'Open ' + this.url + ' wait for page to load';
  }

  async step() {
    await this.page.open(this);
  }
}

module.exports = JoinVideoCallStep;